drainer/relay.go

```diff
@@ -112,8 +120,18 @@ func feedByRelayLog(r relay.Reader, ld loader.Loader, cp checkpoint.CheckPoint)
 			if sbinlog.CommitTs <= checkpointTS {
 				continue
 			}
-
-			txn, err := loader.SecondaryBinlogToTxn(sbinlog)
+			var txn *loader.Txn
+			var err error
+			var tableRouter *router.Table = nil
+			upperColName := false
+			if cfg.SyncerCfg.DestDBType == "oracle" {
+				upperColName = true
+				tableRouter, _, err = genRouterAndBinlogEvent(cfg.SyncerCfg)
+				if err != nil {
+					return errors.Annotate(err, "when feed by relay log, gen router and filter failed")
+				}
+			}
```

Comment:
please put generating `tableRouter` outside this for loop

Reply:
ok